packages/core/src/test/awsService/appBuilder/serverlessLand/main.test.ts

```diff
@@ -0,0 +1,224 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import assert from 'assert'
+import * as sinon from 'sinon'
+import * as path from 'path'
+import { MetadataManager } from '../../../../awsService/appBuilder/serverlessLand/metadataManager'
+import * as main from '../../../../awsService/appBuilder/serverlessLand/main'
+import {
+    createNewServerlessLandProject,
+    openReadmeFile,
+    getProjectUri,
+    downloadPatternCode,
+} from '../../../../awsService/appBuilder/serverlessLand/main'
+import { fs } from '../../../../shared/fs/fs'
+import * as downloadPatterns from '../../../../shared/utilities/downloadPatterns'
+import { ExtContext } from '../../../../shared/extensions'
+import { workspaceUtils } from '../../../../shared'
+import * as downloadPattern from '../../../../shared/utilities/downloadPatterns'
+import * as wizardModule from '../../../../awsService/appBuilder/serverlessLand/wizard'
+
+describe('createNewServerlessLandProject', () => {
+    let sandbox: sinon.SinonSandbox
+    let mockExtContext: ExtContext
+    let mockMetadataManager: sinon.SinonStubbedInstance<MetadataManager>
+    let mockWizard: { run: sinon.SinonStub }
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox()
+        mockExtContext = {
+            awsContext: {
+                getCredentials: sandbox.stub().resolves({}),
+                getCredentialDefaultRegion: () => 'us-west-2',
+            },
+        } as unknown as ExtContext
+
+        mockMetadataManager = sandbox.createStubInstance(MetadataManager)
+        mockMetadataManager.getAssetName.returns('test-asset-name')
+        sandbox.stub(MetadataManager, 'getInstance').returns(mockMetadataManager as unknown as MetadataManager)
```

Comment:
I don't have time to go through all the details. But this feels like we can reduce the amount of stubs needed, just stubbing lower level things like network calls, and then all real code can execute.

Not blocking for this PR due to time constraints, but please keep in mind when designing for testability.

Reply:
I'll cover the changes in a follow up PR